animation_nodes/algorithms/mesh_generation/line.pyx

```diff
@@ -33,10 +41,15 @@ def vertices(start, end, long steps):
 # Edges
 ############################################
 
-def edges(long steps):
-    edges = EdgeIndicesList(length = steps - 1)
-    cdef long i
-    for i in range(steps - 1):
+def edges(long steps, bint cyclic):
+    steps = steps if cyclic and steps > 2 else steps - 1
```

Comment:
Why are you checking for `steps > 2`? This is what you might call a low-level function, it expects valid input and error handling is done at a higher level. Also, it is better if you create a new variable just like you did before instead of writing the the input directly.

Reply:
If cyclic is True and Points are two then it will generate two similar edges which will later create a problem. That is why I have added a check for `steps > 2`.

Reply:
In this case, you can't have two points only when cyclic is enabled. We can raise an exception then.

Reply:
Okay. I'll add an exception in the line.py file.